circuit-benchmarks/src/mpt_circuit.rs

```diff
@@ -0,0 +1,110 @@
+            kzg::{
+                commitment::{KZGCommitmentScheme, ParamsKZG, ParamsVerifierKZG},
+                multiopen::{ProverSHPLONK, VerifierSHPLONK},
+                strategy::SingleStrategy,
+            },
+        },
+        transcript::{
+            Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+        },
+    };
+    use rand::SeedableRng;
+    use rand_xorshift::XorShiftRng;
+    use std::env::var;
+    use zkevm_circuits::mpt_circuit::{witness_row::Node, MPTCircuit};
+
+    #[cfg_attr(not(feature = "benches"), ignore)]
+    #[test]
+    fn bench_mpt_circuit_prover() {
+        let degree: u32 = var("DEGREE")
+            .expect("No DEGREE env var was provided")
+            .parse()
+            .expect("Cannot parse DEGREE env var as u32");
+
+        let path = "../zkevm-circuits/src/mpt_circuit/tests/UpdateOneLevel.json";
+        let file = std::fs::File::open(path.clone());
+        let reader = std::io::BufReader::new(file.unwrap());
+
+        let randomness: Fr = Fr::one() + Fr::one();
+        let nodes: Vec<Node> = serde_json::from_reader(reader).unwrap();
+
+        let mut keccak_data = vec![];
+        for node in nodes.iter() {
+            for k in node.keccak_data.iter() {
+                keccak_data.push(k.clone());
+            }
+        }
+
+        let circuit = MPTCircuit::<Fr> {
+            nodes,
+            keccak_data,
+            randomness,
+        };
+
+        // Initialize the polynomial commitment parameters
+        let mut rng = XorShiftRng::from_seed([
+            0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06,
+            0xbc, 0xe5,
+        ]);
+
+        // Bench setup generation
+        let setup_message = format!("Setup generation with degree = {}", degree);
+        let start1 = start_timer!(|| setup_message);
+        let general_params = ParamsKZG::<Bn256>::setup(degree, &mut rng);
+        let verifier_params: ParamsVerifierKZG<Bn256> = general_params.verifier_params().clone();
+        end_timer!(start1);
+
+        // Initialize the proving key
+        let vk = keygen_vk(&general_params, &circuit).expect("keygen_vk should not fail");
+        let pk = keygen_pk(&general_params, vk, &circuit).expect("keygen_pk should not fail");
+        // Create a proof
+        let mut transcript = Blake2bWrite::<_, G1Affine, Challenge255<_>>::init(vec![]);
+
+        // Bench proof generation time
+        let proof_message = format!("MPT Proof generation with degree = {}", degree);
+        let start2 = start_timer!(|| proof_message);
+
+        create_proof::<
+            KZGCommitmentScheme<Bn256>,
+            ProverSHPLONK<'_, Bn256>,
+            Challenge255<G1Affine>,
+            XorShiftRng,
+            Blake2bWrite<Vec<u8>, G1Affine, Challenge255<G1Affine>>,
+            MPTCircuit<Fr>,
+        >(&general_params, &pk, &[circuit], &[], rng, &mut transcript)
```

Comment:
```suggestion
        >(&general_params, &pk, &[circuit], &[&[]], rng, &mut transcript)
```

I was running the benchmark and the prover time was too fast (especially because some optimizations are not yet on this branch)!

I think it's related to the input here for the instances. If I for example do the same for the keccak circuit it also gets proven much faster. I looked shortly at why in halo2 and it seems like the instance length shouldn't be 0 or some prover work is skipped (though I could be wrong about that). The fact that this doesn't match the input expected by the MockProver input is pretty confusing.

However, if I change it as suggested I get a `ConstraintSystemFailure`... Not sure what's up yet.